pkg/controller/vitessshard/vitessshard_controller.go

```diff
@@ -171,6 +171,8 @@ func (r *ReconcileVitessShard) Reconcile(request reconcile.Request) (reconcile.R
 	// Reset status, since that's all out of date info that we will recompute now.
 	oldStatus := vts.Status
 	vts.Status = planetscalev2.NewVitessShardStatus()
+	// Make sure we don't rinse old conditions - these states need to persist.
```

Comment:
Add justification for why we need to persist these: because they may have been added by some other controller.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/31/commits/448517eb5d5944f4f2749ee076669f504707f29b.